processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -303,16 +322,48 @@ protected IncrementalIndex(
       DimensionHandler handler = DimensionHandlerUtils.getHandlerFromCapabilities(
           dimName,
           capabilities,
-          dimSchema.getMultiValueHandling()
+          dimSchema.getMultiValueHandling(),
+          enableInMemoryBitmap
       );
-      addNewDimension(dimName, handler);
+      DimensionDesc desc = addNewDimension(dimName, handler);
+
+      if (enableInMemoryBitmap && type.equals(ColumnType.STRING)) {
```

Comment:
Why only String dimension is supported?

Reply:
Good catch!

https://github.com/apache/druid/blob/master/docs/ingestion/ingestion-spec.md
in dimensionSpec, createBitmapIndex (default to true)
```
For string typed dimensions, whether or not bitmap indexes should be created for the column in generated segments. Creating a bitmap index requires more storage, but speeds up certain kinds of filtering (especially equality and prefix filtering). Only supported for string typed dimensions.	
```

So it means currently bitmap index in batch immutable segments only support string type columns, in this PR I try to follow the same design by still only supporting string typed dimensions but extending the bitmap index support from batch immutable segments to real-time incremental index. In-memory bitmaps will be enabled if both `createBitmapIndex` and `enableInMemoryBitmap` are true. 

Not sure if it makes sense to have any such case: `createBitmapIndex` is false (bitmap index is disabled in batch immutable segments) and `enableInMemoryBitmap` is true (bitmap index is enabled in incremental index), to avoid confusion, I avoided this case by disabling in-memory bitmap in this case. Let me know if you have other thoughts.

I'll also document this in detail.